pilot/pkg/networking/core/v1alpha3/thriftroute.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha3
+
+import (
+	route "github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
+	thrift_proxy "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/thrift_proxy/v2alpha1"
+
+	"istio.io/istio/pilot/pkg/features"
+)
+
+// buildDefaultThriftInboundRoute builds a default inbound route.
+func buildDefaultThriftRoute(clusterName string) *thrift_proxy.Route {
+	var rateLimits []*route.RateLimit
+	if features.ThriftRatelimitService.Get() != "" {
+		rateLimits = []*route.RateLimit{
+			{
+				Actions: []*route.RateLimit_Action{
+					{
+						ActionSpecifier: &route.RateLimit_Action_SourceCluster_{
+							// Automatically populated
+							SourceCluster: &route.RateLimit_Action_SourceCluster{},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	return &thrift_proxy.Route{
+		Match: &thrift_proxy.RouteMatch{
+			MatchSpecifier: &thrift_proxy.RouteMatch_MethodName{
+				MethodName: "",
```

Comment:
Based on this it looks like you can only have a single thrift service for each port, cluster wide? Is that correct or am I missing something here?

Reply:
If I understand correctly this will match on service + port pair, meaning we'll have the entire port range per service.

Reply:
Ah right, we are listening on svcIP:port. Got it. I was thinking of http where we listen on 0.0.0.0:port and route based on host header